src/server/pfs/server/driver.go

```diff
@@ -130,10 +131,22 @@ func newDriver(
 		return nil, errors.Errorf("cannot initialize driver with nil treeCache")
 	}
 	// Initialize driver
+	objClient, err := obj.NewClientFromSecret(storageRoot)
+	if err != nil {
+		return nil, err
+	}
+	if env.StorageDiskCacheSize > 0 {
+		diskCache, err := obj.NewLocalClient(filepath.Join(os.TempDir(), "pfs-cache", uuid.NewWithoutDashes()))
+		if err != nil {
+			return nil, err
+		}
+		objClient = obj.NewCacheClient(objClient, diskCache, env.StorageDiskCacheSize)
```

Comment:
Yeah, I think this is a fair concern, but I don't think trying to unify the caches in 1.x is going to be worth the effort. Right now we essentially have three PFS caches: object, input repo metadata, and output repo metadata. The object one takes into account size when caching and the input / output repo metadata ones don't. I think this is fine for now since in general the metadata will be much smaller than the data, and in 2.0 we don't need to account for the size of individual chunks because we can bound it based on the max chunk size configuration (which is why `env.StorageDiskCacheSize` is for the number of objects). Also, the output repo metadata objects that are cached in general will be smaller than the input repo ones since it is only caching a subset of metadata (path range) not a full hashtree (boltDB), which is why that default is much higher than the default for the input repo metadata cache.